packages/legend-graph/src/graph/helpers/DomainHelper.ts

```diff
@@ -230,11 +230,21 @@ export const isMainGraphElement = (
 export const isElementReadOnly = (element: PackageableElement): boolean =>
   !isMainGraphElement(element);
 
+// Each dependency project is considered as a root. Dependency project name is `${groupId}:${artifactId}`
+// So dependency name contains ':'. Here we are making the assumption that only dependecy project name
+// has this exception and we are making the check for dependency element on basis of that.
 export const isDependencyElement = (
```

Comment:
Not really that excited for this check here. 
Could we maybe have pureModel been an input to this function as well and check if the root package of the element is one of the root packages in dependency manager of the pure model?

Reply:
makes sense. Will modify that